packages/media-server/srv/media-service.js

```diff
@@ -0,0 +1,82 @@
+module.exports = (srv) => {
+    const loki = require('lokijs');
+    const db = new loki('DB');
+    const mediaDB = db.addCollection("Media");
+    const { Readable, PassThrough } = require('stream')
+
+    srv.before('CREATE', 'Media', (req) => {
+        const obj = mediaDB.insert({ media: "" });
+        req.data.id = obj.$loki;
+    })
+
+    srv.on('UPDATE', 'Media', (req, next) => {
+        try{ 
+            const url = req._.req.path;
+            if (url.indexOf("content")>0) {
+                const id = req.data.id; 
+                const obj = mediaDB.get(id);
+                if(obj==null) {
+                    req.reject(404,"No record found for the ID");
+                    return;
+                }
+                const stream = new PassThrough();
+                let chunks = [];
+                stream.on('data',  (chunk) => {
+                    chunks.push(chunk);
+                });
+                stream.on('end',  () =>{
+                    obj.media = Buffer.concat(chunks).toString('base64');
+                    mediaDB.update(obj);
+                });
+                req.data.content.pipe(stream);
+            }  else return next()
+        }catch(error){
+            req.reject(404,"Media DB error")
```

Comment:
The error `404` does not reflect a `Media DB error`. Do not catch the error, let it go through, it will result in a (correct) error code `500` (internal server error).

Reply:
Have removed the try/catch